plugins/modules/ndo_ipsla_track_list.py

```diff
@@ -0,0 +1,468 @@
+                            schema=dict(type="str"),
+                        ),
+                    ),
+                    scope_type=dict(type="str", choices=["bd", "l3out"], required=True),
+                    weight=dict(type="int"),
+                ),
+            ),
+            state=dict(type="str", choices=["absent", "query", "present"], default="present"),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["ipsla_track_list", "ipsla_track_list_uuid"], True],
+            ["state", "present", ["ipsla_track_list", "ipsla_track_list_uuid"], True],
+            ["state", "present", ["type"]],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    ipsla_track_list = module.params.get("ipsla_track_list")
+    description = module.params.get("description")
+    ipsla_track_list_uuid = module.params.get("ipsla_track_list_uuid")
+    list_type = module.params.get("type")
+    thresholds = {
+        "down": module.params.get("threshold_down"),
+        "up": module.params.get("threshold_up"),
+    }
+    members = module.params.get("members")
+    state = module.params.get("state")
+
+    # Validate
+    valid_upper = 100
+    valid_lower = 0
+    if list_type == "weight":
+        valid_upper = 255
+    for threshold_key, threshold_value in thresholds.items():
+        if threshold_value is not None and threshold_value not in range(valid_lower, valid_upper):
+            mso.fail_json(
+                msg="Invalid value provided for threshold_{0}: {1}; it must be in the range {2} - {3}".format(
+                    threshold_key, threshold_value, valid_lower, valid_upper
+                )
+            )
+
+    ops = []
+    match = None
+
+    # The object dictionary is used as a cache store for schema & template data.
+    # This is done to limit the amount of API calls when UUID is not specified for member scope references.
+    obj_cache = {}
+
+    mso_template = MSOTemplate(mso, "tenant", template)
+    mso_template.validate_template("tenantPolicy")
+    obj_cache["template-tenant-{0}".format(template)] = mso_template
+
+    object_description = "IPSLA Track List"
+    path = "/tenantPolicyTemplate/template/ipslaTrackLists"
+
+    existing_ipsla_track_lists = mso_template.template.get("tenantPolicyTemplate", {}).get("template", {}).get("ipslaTrackLists", [])
+    if ipsla_track_list or ipsla_track_list_uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_ipsla_track_lists,
+            [(KVPair("uuid", ipsla_track_list_uuid) if ipsla_track_list_uuid else KVPair("name", ipsla_track_list))],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_ipsla_track_lists
```

Comment:
Insert the `ipslaMonitoringRef` and `scope` object name in the `current, previous` output portion when we do the `create, update, query, delete`.